### PR TITLE
Add gotify reporter

### PR DESCRIPTION
**Pull request by franco-righetti**
I've tested the reporter with `--test-reporter gotify` and also the default configuration generation with `--edit-config`.

### CODE REVIEW

File: docs/source/reporters.rst [hunk 1]
```diff
@@ -210,6 +211,30 @@ Embedded content might be easier to read and identify individual reports. Subjec
 
 When `colored` is true reports will be embedded in code section (with diff syntax) to enable colors.
 
+Gotify
+------
+
```
Comment (thp) [inline docs/source/reporters.rst:216]: Maybe link to https://gotify.net and/or at least say what Gotify is?

**Review by thp — CHANGES_REQUESTED** [review]
This looks good. Changed 2**14 to 16 * 1024 for easier human readability.

Please add a line to the `CHANGELOG.md` file describing the change.

**Comment by thp** [comment]
CI failures fixed in https://github.com/thp/urlwatch/commit/556145906cddea2e723bf489073bfeef638c82c0 in the master branch, if you merge the upstream master branch into your PR branch, the CI should fix itself.

File: docs/source/reporters.rst (hunk 1 above)
Comment (franco-righetti) [reply docs/source/reporters.rst:216]: Short description with a link added.

**Comment by franco-righetti** [comment]
Requested changes have been made. Sorry for the delay.

**Review by thp — APPROVED** [review]
Thanks!

**Pull request merged**